Classes/DDFileLogger.m

```diff
@@ -1077,27 +1135,26 @@ - (void)willLogMessage {
 }
 
 - (void)didLogMessage {
-    [self maybeRollLogFileDueToSize];
+    [self lt_maybeRollLogFileDueToSize];
 }
 
 - (BOOL)shouldArchiveRecentLogFileInfo:(__unused DDLogFileInfo *)recentLogFileInfo {
     return NO;
 }
 
 - (void)willRemoveLogger {
-    // If you override me be sure to invoke [super willRemoveLogger];
+    [self lt_rollLogFileNow];
+}
 
-    [self rollLogFileNow];
+- (void)flush {
+    NSAssert([self isOnInternalLoggerQueue], @"lt_ methods should be on logger queue.");
```

Comment:
Assertion method is incorrect, it's not an `lt_` method...

Reply:
This method is currently only called from inside `DDLog lt_flush`, which dispatches onto the logger's queue before calling the method. I was considering renaming it, but I believe that would be another breaking change since the method is defined in the public `DDLogger` protocol. I therefore opted to insert the check and not rename the method. I'll insert a comment since this is rather non-intuitive.

Reply:
A comment would most certainly help, yea.
But why deprecate `flush` in your alternative? Why not just add `lt_flush` and forward `flush` -> `lt_flush`? People calling `flush` from outside the logger probably *want* it to be safe (as in dispatched to the internal queue), right?

Reply:
Yea, let's make flush save for outside use 👍